bdk_chain/src/keychain/keychain_tracker.rs

```diff
@@ -285,13 +296,34 @@ where
             .map(|(_, full_txout)| full_txout.txout.value)
             .sum()
     }
+
+    /// Marks a UTXO(`outpoint`) as selected by adding it to a set of utxos used
+    /// by transactions that have been created but not yet broadcasted.
+    /// Once a utxo is marked as selected by a particular transaction either
+    /// during/after the transaction building process, it cannot be used for
+    /// building another transaction.
+    /// This is useful in a scenario where you are creating a sequence of
+    /// transactions and you don't want utxos to be re-used as inputs
+    /// across transactions.
+    /// A UTXO will be considered selected until it is released by [`Self::unmark_selected`]
+    /// to the wallet's UTXO pool.
```

Comment:
Nit: I think it will be better if this was split into "sections" as mentioned here: https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components